connector_magento/models/sale_order/importer.py

```diff
@@ -715,8 +715,9 @@ def _import_dependencies(self):
                     key = 'product_id'
                 else:
                     key = 'sku'
-                self._import_dependency(line[key],
-                                        'magento.product.product')
+            else:
+                key = 'sku'
+            self._import_dependency(line[key], 'magento.product.product')
```

Comment:
Please use this logic instead (from lower down this file): https://github.com/OCA/connector-magento/blob/12.0/connector_magento/models/sale_order/importer.py#L750-L753 and change the check to `if key in line`.